src/main/java/cinema/dao/hibernateimpl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package cinema.dao.hibernateimpl;
+
+import cinema.dao.MovieSessionDao;
+import cinema.lib.Dao;
+import cinema.model.MovieSession;
+import cinema.util.HibernateUtil;
+import java.time.LocalDate;
+import java.util.List;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long movieSessionId = (Long) session.save(movieSession);
+            transaction.commit();
+            movieSession.setId(movieSessionId);
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert MovieSession entity", e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<MovieSession> movieSessionCriteriaQuery =
```

Comment:
```suggestion
            CriteriaQuery<MovieSession> criteriaQuery =
```
оскільки інших criteriaQuery тут нема, такої назви цілком достатньо